rixs/process2d.py

```diff
@@ -0,0 +1,550 @@
+
+    kwargs.update({key: val for key, val in defaults.items()
+                   if key not in kwargs})
+
+    xlim = (x_edges.min(), x_edges.max())
+    ylim = (y_edges.min(), y_edges.max())
+    image_artist = ax.pcolorfast(xlim, ylim, image, **kwargs)
+    cb_artist = plt.colorbar(image_artist, ax=ax, cax=cax)
+    ax.axis('tight')
+
+    return image_artist, cb_artist
+
+
+def poly(x, x2=0., x1=0., x0=500.):
+    """Second order polynominal function for fitting curvature.
+    Returns x2*x**2 + x1*x + x0
+    """
+    return x2*x**2 + x1*x + x0
+
+
+def get_curvature_offsets(photon_events, bins=None):
+    """ Determine the offests that define the isoenergetic line.
+    This is determined as the maximum of the cross correlation function with
+    a reference taken from the center of the image.
+
+    Parameters
+    ------------
+    photon_events : array
+        three column x, y, Iph photon locations and intensities
+    bins : int or array_like or [int, int] or [array, array]
+        The bin specification in y then x order:
+            * If None
+            * If int, the number of bins for the two dimensions (nx=ny=bins).
+            * If array_like, the bin edges for the two dimensions
+              (y_edges=x_edges=bins).
+            * If [int, int], the number of bins in each dimension
+              (ny, nx = bins).
+            * If [array, array], the bin edges in each dimension
+              (y_edges, x_edges = bins).
+            * A combination [int, array] or [array, int], where int
+              is the number of bins and array is the bin edges.
+
+    Returns
+    -------------
+    x_centers : array
+        columns positions where offsets were determined
+        i.e. binx/2, 3*binx/2, 5*binx/2, ...
+    offsets : array
+        np.array of row offsets defining curvature. This is referenced
+        to the center of the image.
+    """
+    x = photon_events[:, 0]
+    y = photon_events[:, 1]
+    Iph = photon_events[:, 2]
+
+    if bins is None:
+        ybins = np.arange(y.min()//1 + 0.5, y.max()//1, 1)
+        xstep = 50
+        xbins = np.arange(x.min() + xstep/2, x.max() - xstep/2, xstep)
+        bins = (ybins, xbins)
+
+    image, y_edges, x_edges = np.histogram2d(y, x, bins=bins, weights=Iph)
+    y_centers = (y_edges[:-1] + y_edges[1:])/2
+    x_centers = (x_edges[:-1] + x_edges[1:])/2
+
+    ref_column = image[:, image.shape[1]//2]
+
+    offsets = np.array([])
+    for column in image.T:
+        cross_correlation = np.correlate(column, ref_column, mode='same')
+        offsets = np.append(offsets, y_centers[np.argmax(cross_correlation)])
+    return x_centers, offsets - offsets[offsets.shape[0]//2]
+
+
+def fit_curvature(photon_events, bins=None,
+                  guess_curvature={'x2': 0., 'x1': 0., 'x0': 500.},
+                  vary_params={'x2': True, 'x1': True, 'x0': True}):
+    """Get offsets, fit them and return polynomial that defines the curvature
```

Comment:
Again a longer doc string would help in understanding what this does.

Also is there a way to write this so that it takes in an arbitrary function and fits that (instead of assuming a polynomial). I think this will allow for greater flexibility later.

Reply:
OK based on the correspondence above about polynomial fitting, perhaps the best option is to use numpy.polyfit in place of this function.